core/src/apps/management/change_language.py

```diff
@@ -13,6 +13,10 @@
 _CHUNK_SIZE = const(1024)
 
 
+class AllocationFailed(MemoryError):
```

Comment:
don't see why we need a separate exception? you may as well handle `MemoryError` in the try-except block below

Reply:
I have added a separate exception just to be sure we skip an OOM originating from blob allocation (and not some other allocation done in the rest of the code), but indeed it is better to ignore any `MemoryError` originating the language change flow during firmware upgrade.

Reply:
82e68d7e0ad7514c70c41fb1a2bff1bb051c0ede